test/functional/jar.test.ts

```diff
@@ -0,0 +1,31 @@
+import * as  test from 'tap-only';
+import * as path from 'path';
+import {getSha1ForJar, isJar} from '../../lib/jar';
+
+test('detects jar files', (t) => {
+  const jarPaths = [
+    'mvn-app-1.0-SNAPSHOT.jar',
+    'mvn-app-1.0-SNAPSHOT.war',
+    'mvn-app-1.0-SNAPSHOT.zip'];
+  for (const path of jarPaths) {
+    t.ok(isJar(path), path);
+  }
+  t.end();
```

Comment:
why do we need to call `t.end()`? just curious

Reply:
it signals to tap that the test has finished when calling asynchronous functions https://node-tap.org/docs/getting-started/#async-stuff
we don't actually need this here @orsagie because isJar is synchronous

Reply:
test was failing without it.

Reply:
I believe if you fo a `forEach` and pass `t` down it will be fine